### DISCUSSION TITLE
@Property() with a name starting with __ are not included by default if no serialization groups are specified

### DISCUSSION THREAD

**Discussion by SteadEXE** [mikro-orm/mikro-orm#5731] [General]
### Describe the bug

One of my entity has a virtual field
```typescript
// Property to mimic graphql special __typename so it does not break authorization library
@Property({ persist: false, name: '__typename' })
__typename() {
  return File.name;
}
```
If I call serialize() on this entity, it doesn't serialize this field unless I specify a group in the @Property()

### Reproduction

Add the following on an entity and try to call serialize() on it passing a group.
```typescript
@Property({ persist: false, name: '__typename' })
__typename() {
  return File.name;
}
```

### What driver are you using?

@mikro-orm/mariadb

### MikroORM version

6.2.9

### Node.js version

v20.14.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
No properties with `_` prefix are included, a single underscore is enough, and this works the same with implicit and explicit serialization. 

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/serialization/EntityTransformer.ts#L14
https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/serialization/EntitySerializer.ts#L42

This has been there for years, changing it feels pretty breaking to me, so maybe in v7.

**Reply by SteadEXE** [reply]
Sounds clear for me, yeah it might be breaking existing stuff.

**Reply by B4nan** [reply]
As you can see in the code itself, you can also explicitly mark those properties as populated to have them included, groups are another way to get around this.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/serialization/EntitySerializer.ts#L33

We could have a flag to skip this and flip the switch in v7 I guess.

Alternatively, you could drop the `__` prefix from your property name and only use it in `serializedName`.

https://mikro-orm.io/docs/serializing#property-serializers
